src/components/renderer/field/form-field-renderer.component.tsx

```diff
@@ -172,6 +172,18 @@ export const FormFieldRenderer = ({ fieldId, valueAdapter, repeatOptions }: Form
       />
     );
   }
+
+  // If the field is transient and has no value, we do not render it in embedded view mode.
+  // This is to prevent transient fields from being displayed in the form when they are not necessarily needed.
+  if (
+    sessionMode === 'embedded-view' &&
+    field.questionOptions.isTransient &&
```

Comment:
```suggestion
    field?.questionOptions?.isTransient &&
```

Reply:
@gitcliff It's null-safe to access the `isTransient` because the `questionOptions` object will never be null or undefined at this point.